packages/framework/src/validators/class-validator.validator.ts

```diff
@@ -0,0 +1,151 @@
+    // eslint-disable-next-line no-param-reassign
+    schema.additionalProperties = false;
+  }
+
+  if (schema.properties) {
+    for (const key in schema.properties) {
+      if (schema.properties.hasOwnProperty(key)) {
+        addAdditionalPropertiesFalse(schema.properties[key]);
+      }
+    }
+  }
+
+  return schema;
+}
+function formatErrors(errors: ValidationError[], parentPath = ''): { path: string; message: string }[] {
+  return errors.flatMap((err) => {
+    const currentPath = `${parentPath}/${err.property}`.replace(/\/+/g, '/');
+
+    if (err.children && err.children.length > 0) {
+      // Recursively format the children
+      return formatErrors(err.children, currentPath);
+    } else {
+      // Base case: no children, return the formatted error
+      return {
+        path: currentPath,
+        message: Object.values(err.constraints || {}).join(', '),
+      };
+    }
+  });
+}
+
+export class ClassValidatorValidator implements Validator<ClassType> {
+  canHandle(schema: Schema): schema is ClassType {
+    return typeof schema === 'function' && schema.prototype !== undefined && schema.prototype.constructor === schema;
+  }
+
+  async validate<
+    T_Schema extends ClassType = ClassType,
+    T_Unvalidated = FromSchemaUnvalidated<T_Schema>,
+    T_Validated = FromSchema<T_Schema>,
+  >(data: T_Unvalidated, schema: T_Schema): Promise<ValidateResult<T_Validated>> {
+    try {
+      await import('reflect-metadata');
+      const { plainToInstance, instanceToPlain } = await import('class-transformer');
+      const { validate } = await import('class-validator');
+
+      // Convert plain data to an instance of the schema class
+      const instance = plainToInstance(schema, data);
+
+      // Validate the instance
+      const errors = await validate(instance as object, { whitelist: true });
+
+      // if undefined, then something went wrong
+      if (!instance && !!data) throw new Error('Failed to convert data to an instance of the schema class');
+
+      if (errors.length === 0) {
+        return { success: true, data: instanceToPlain(instance) as T_Validated };
+      } else {
+        return {
+          success: false,
+          errors: formatErrors(errors),
+        };
+      }
+    } catch (error) {
+      if ((error as Error)?.message?.includes('Cannot find module')) {
+        throw new Error(
+          'Tried to use a class-validator schema in @novu/framework without `class-validator`, `class-transformer` or `reflect-metadata` installed. ' +
+            'Please install it by running `npm install class-validator class-transformer reflect-metadata`.'
+        );
+      }
+      throw error;
+    }
+  }
+
+  async transformToJsonSchema(schema: ClassType): Promise<JsonSchema> {
+    try {
+      /*
+       * TODO: replace with direct import, when metadataStorage is exported by default: https://github.com/typestack/class-transformer/issues/563#issuecomment-803262394
+       * @ts-ignore
```

Comment:
It's preferable to use `ts-expect-error` directives over `ts-ignore` to become aware of when a downstream change fixes an issue.

```suggestion
       * @ts-expect-error - class-transformer doesn't export `defaultMetadataStorage` from the root module
```